File: Scripts/results/generated_code_results/Flutter/Education/WiseOnlineCourses/DownloadsScreen.dart
import 'package:flutter/material.dart';
class DownloadsScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
backgroundColor: Color(0xFF1E1E1E),
appBar: AppBar(
backgroundColor: Colors.transparent,
elevation: 0,
title: Text(
'Downloads',
style: TextStyle(color: Colors.white, fontSize: 24),
),
actions: [
Padding(
padding: const EdgeInsets.all(8.0),
child: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
),
],
),
body: ListView(
children: [
DownloadItem(
title: 'Gift yourself this UI Kit and enjoy!',
instructor: 'John Wiseberg',
duration: '08:12',
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
),
DownloadItem(
title: 'Make your Mess Your Message',
instructor: 'Mendy Santiago',
duration: '12:38',
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
),
DownloadItem(
title: 'Meet Your Dreams and Make It Real',
instructor: 'Maria Trofimova',
duration: '09:46',
onTap: () => Navigator.pushNamed(context, 'CourseDetail'),
),
],
),
bottomNavigationBar: BottomNavigationBar(
backgroundColor: Color(0xFF1E1E1E),
selectedItemColor: Colors.white,
unselectedItemColor: Colors.grey,
currentIndex: 2,
items: [
BottomNavigationBarItem(
icon: Icon(Icons.home),
label: 'Home',
),
BottomNavigationBarItem(
icon: Icon(Icons.search),
label: 'Search',
),
BottomNavigationBarItem(
icon: Icon(Icons.file_download),
label: 'Downloads',
),
],
onTap: (index) {
if (index == 0) {
Navigator.pushNamed(context, 'Home');
}
},
),
);
}
}
class DownloadItem extends StatelessWidget {
final String title;
final String instructor;
final String duration;
final VoidCallback onTap;
const DownloadItem({
Key? key,
required this.title,
required this.instructor,
required this.duration,
required this.onTap,
}) : super(key: key);
@override
Widget build(BuildContext context) {
return GestureDetector(
onTap: onTap,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 16, vertical: 12),
child: Row(
children: [
Container(
width: 120,
height: 80,
decoration: BoxDecoration(
borderRadius: BorderRadius.circular(8),
image: DecorationImage(
image: AssetImage('assets/images/template.png'),
fit: BoxFit.cover,
),
),
child: Align(
alignment: Alignment.bottomRight,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 6, vertical: 2),
decoration: BoxDecoration(
color: Colors.black.withOpacity(0.7),
borderRadius: BorderRadius.circular(4),
),
child: Text(
duration,
style: TextStyle(color: Colors.white, fontSize: 12),
),
),
),
),
SizedBox(width: 16),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
title,
style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
),
SizedBox(height: 4),
Text(
instructor,
style: TextStyle(color: Colors.grey, fontSize: 14),
),
],
),
),
],
),
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Education/WiseOnlineCourses/DownloadsScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, ImageBackground, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';
import { useNavigation } from '@react-navigation/native';

const DownloadsScreen = () => {
  const navigation = useNavigation();

  const handleCourseDetailNavigation = (item) => {
    navigation.navigate('CourseDetail', { item });
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.headerTitle}>Downloads</Text>
        <Image style={styles.avatar} source={require('../assets/snack-icon.png')} />
      </View>
      <View style={styles.downloadList}>
        {downloadItems.map((item, index) => (
          <TouchableOpacity key={index} style={styles.downloadItem} onPress={() => handleCourseDetailNavigation(item)}>
            <ImageBackground style={styles.thumbnail} source={require('../assets/snack-icon.png')}>
              <Text style={styles.duration}>{item.duration}</Text>
            </ImageBackground>
            <View style={styles.textContainer}>
              <Text style={styles.title}>{item.title}</Text>
              <Text style={styles.instructor}>{item.instructor}</Text>
            </View>
          </TouchableOpacity>
        ))}
      </View>
      <View style={styles.bottomNav}>
        <Icon name="home" style={styles.icon} />
        <Icon name="search" style={styles.icon} />
        <Icon name="download" style={styles.iconActive} />
      </View>
    </View>
  );
};

const downloadItems = [
  { title: 'Gift yourself this UI Kit and enjoy!', instructor: 'John Wiseberg', duration: '08:12' },
  { title: 'Make your Mess Your Message', instructor: 'Mendy Santiago', duration: '12:38' },
  { title: 'Meet Your Dreams and Make It Real', instructor: 'Maria Trofimova', duration: '09:46' }
];

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#1a1c23',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    margin: 20,
  },
  headerTitle: {
    color: '#fff',
    fontSize: 24,
    fontWeight: 'bold',
  },
  avatar: {
    width: 40,
    height: 40,
    borderRadius: 20,
  },
  downloadList: {
    flex: 1,
  },
  downloadItem: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10,
    marginHorizontal: 20,
  },
  thumbnail: {
    width: 100,
    height: 70,
    justifyContent: 'flex-end',
  },
  duration: {
    color: '#fff',
    fontSize: 12,
    backgroundColor: 'rgba(0,0,0,0.7)',
    padding: 4,
    textAlign: 'center',
  },
  textContainer: {
    flex: 1,
    marginLeft: 15,
  },
  title: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
  instructor: {
    color: '#888',
    fontSize: 14,
    marginTop: 4,
  },
  bottomNav: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    backgroundColor: '#111217',
    height: 60,
    position: 'absolute',
    bottom: 0,
    width: '100%',
  },
  icon: {
    color: '#888',
    fontSize: 24,
  },
  iconActive: {
    color: '#fff',
    fontSize: 24,
  },
});

export default DownloadsScreen;